logger/include/logger.hpp

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include <filesystem>
+#include <iostream>
+#include <list>
+#include <source_location>
+#include <sstream>
+
+enum Loglevel
+{
+    INFO,
+    CRITICAL
+};
+
+namespace Logger
+{
+
+std::list<std::string> buffer;
+const size_t size = 8000;
+
+/**
+ *@brief Appends logmessage to the buffer
+ *
+ *This API is called to log message to the buffer.
+ *It checks the log level and appends information like
+ *line number, file name, function name to the buffer.
+ *
+ *@param[in] - loglevel of the message
+ *@param[ni] - Information to be logged
+ *@param[in] - object of the source_location class
+ */
+void logmessage(
+    Loglevel level, const std::string message,
+    const std::source_location& location = std::source_location::current())
+{
+    if (Loglevel::INFO == level)
+    {
+        buffer.push_back(
+            " line no: " + std::to_string(location.line()) + " File: " +
+            std::filesystem::path(location.file_name()).filename().string() +
+            " Function Name: " + location.function_name() + " " + message +
+            "\n");
+    }
+
+    while (buffer.size() > size)
```

Comment:
why while() loop is required here? why can't just be an if() condition. 
I think the above buffer.push_back() pushes one line at a time, is that right?